### ISSUE TITLE
POSITION_TARGET_LOCAL_NED stream not appearing 

### ISSUE THREAD

**Issue by protobits** [PX4/PX4-Autopilot#12757]
**Describe the bug**
Quadcopter set to altitude mode (on ground, disarmed) with PX4FLOW sensor attached. No POSITION_TARGET_LOCAL_NED stream on QGC visible

**To Reproduce**
1. Quadcopter with PX4FLOW attached, disarmed
2. Altitude mode set via radio transmitter (confirmed in QGC)
3. No POSITION_TARGET_LOCAL_NED stream visible in QGC analyze

**Expected behavior**
I expect to be able to switch to altitude mode and observe the setpoint in order to tune the altitude controller. Same thing happens if set to position control with local position input from onboard computer. I'm using FMUv2 board. I tried observing the visible mavlink streams from TELEM1 (set to "Normal" as per default) and from USB port. PX4 version is 1.9.2.

**Log Files and Screenshots**
No log as the drone was not armed.

Add screenshots to help explain your problem.

**Drone (please complete the following information):**
- Custom built quadcopter
- Pixhawk FMUv2 with PX4FLOW sensor attached via I2C

**Comment by julianoes** [comment]
I think you would need to request the MAVLink message stream using:
https://mavlink.io/en/messages/common.html#MAV_CMD_SET_MESSAGE_INTERVAL

**Comment by v01d** [comment]
According to the code I've read, this stream is supposed to be on for "normal" and "onboard" stream types.

**Comment by julianoes** [comment]
Ok, and what mode are you using?

**Comment by v01d** [comment]
This is on default TELEM1 port with Normal mode (the default).

**Comment by julianoes** [comment]
I'm trying to reproduce this and I can see `POSITION_TARGET_LOCAL_NED` being streamed at around 0.5 Hz when flying in HITL in MISSION, POSCTL, and ALTCTL.

However, the values in ALTCTL are all `NaN` except `vz`.

![Selection_035](https://user-images.githubusercontent.com/1419688/63497745-bcb31d80-c4c4-11e9-9842-2e2afb12595f.png)


In POSCTL everything seems sane, except `z` is still `NaN`. I wonder why that happens, presumably it's just not passed to this message.

![Selection_036](https://user-images.githubusercontent.com/1419688/63497753-c177d180-c4c4-11e9-9af3-ffb516a60057.png)

**Comment by julianoes** [comment]
@bresch it looks like the z setpoint is `NaN` for ALTCTL and POSCTL. I think that's not intuitive and makes it weird to tune.

**Comment by julianoes** [comment]
The issue where the z setpoint is NaN is a recent problem -> #12776.

**Comment by v01d** [comment]
Maybe it is because of this that the mavlink message is not visible (some not Nan guard avoid publication)?

**Comment by julianoes** [comment]
> Maybe it is because of this that the mavlink message is not visible (some not Nan guard avoid publication)?

Let me check!

**Comment by julianoes** [comment]
This works for me now with latest master (the screenshot is SITL) but this should be the same thing over telem1 just only 0.5 Hz.

![Selection_037](https://user-images.githubusercontent.com/1419688/63524623-a032d780-c4fc-11e9-9207-882e63676268.png)

**Comment by v01d** [comment]
I've flashed current master on a standalone Pixhawk board (just RC receiver connected) and, while I can switch to altitude mode, I still don't get position target stream (neither on USB port nor TELEM1 port):

![image](https://user-images.githubusercontent.com/161706/63535957-09105480-c4e9-11e9-8984-1faae05b0d2a.png)

**Comment by julianoes** [comment]
Ok, and that's with a GPS lock?

**Comment by v01d** [comment]
No, no lock. This is a board with only RC attached. It is on altitude mode using baro.

**Comment by julianoes** [comment]
This might be the issue then. Can you check if you see the message with GPS?

**Comment by julianoes** [comment]
> Do you mean that the setpoint will not be visible without GPS? In altitude mode I believe it should appear even without GPS since it is sending an altitude setpoint to follow with barometer. Right?

I agree that it should but first I need to understand why it doesn't.
